lib/shopify_app/controller_concerns/with_shopify_id_token.rb

```diff
@@ -5,31 +5,30 @@ module WithShopifyIdToken
     extend ActiveSupport::Concern
 
     def shopify_id_token
-      @shopify_id_token ||= id_token_from_request_env || id_token_from_authorization_header || id_token_from_url_param
+      @shopify_id_token ||= id_token_from_authorization_header || id_token_from_url_param
```

Comment:
If there's no token in header or URL param, this will memoize nil and keep evaluating the `id_token_from_authorization_header || id_token_from_url_param` part on future calls.

Perhaps we should write this as

```suggestion
      return @shopify_id_token if defined?(@shopify_id_token)
      @shopify_id_token = id_token_from_authorization_header || id_token_from_url_param
```

Reply:
(there are similar concerns with the memoization in the other methods here, not sure how important this is).